frontend/src/metabase/parameters/utils/mapping-options.ts

```diff
@@ -28,13 +50,17 @@ function buildStructuredQuerySectionOptions(query, stageIndex, group) {
       sectionName: getColumnGroupName(groupInfo),
       name: columnInfo.displayName,
       icon: getColumnIcon(column),
-      target: buildColumnTarget(query, stageIndex, column),
+      target: buildColumnTarget(
+        query,
+        stageIndex,
+        column,
+      ) as ParameterDimensionTarget,
```

Comment:
I think this cast hides some typing problems.
We should explicitly specify the return type of `buildColumnTarget` to be `ParameterDimensionTarget` and go from there.

Reply:
that was my initial thought, but it didn't work out. Let me double check